Makefile

```diff
@@ -28,17 +28,16 @@ help: ## Display this help message
 	    awk -F ':.*?## ' 'NF==2 {printf "  %-26s%s\n\n", $$1, "${COLOUR_GREEN}"$$2"${COLOUR_END}"}'
 
 .PHONY: setup
-setup: setup-pre-commit check-radoslib setup-venv ## Setup the pre-commit environment and then the venv environment
+setup: check-radoslib setup-venv setup-pre-commit ## Setup the pre-commit environment and then the venv environment
 
 .PHONY: setup-pre-commit
 setup-pre-commit:
-	pip install --user pre-commit && pre-commit install
+	./.venv/bin/pip install --user pre-commit && ./.venv/bin/python3.12 -m pre_commit install
 
```

Comment:
virtualenv access should not be version dependent

Reply:
but the gitlab ci is version dependent and it is running on python3.12. So should we then make ci dependent on a python version? Should rookify itself not be dependent on a specific python version?

Reply:
ok I now added some dynamic python check to add the python executable (ubuntu uses `python3` or `python2` not `python` unless you install the package `python-is-python3` haha